src/routers/products.router.ts

```diff
@@ -5,18 +5,24 @@ import {
   createProduct,
   getPopularInTheCommunity,
   uploadProductImage,
-  getLimitedEdtionProducts
+  getLimitedEdtionProducts,
+  searchProducts
 } from '../controllers/products.controller';
 import { use } from '../helpers';
 import { paginateMiddleware } from '../middlewares/paginate.middleware';
 
 const productsRouter: Router = Router();
 
+productsRouter.use(
+  ['/popular', '/limited-edition', '/', '/search'],
+  paginateMiddleware
+);
 productsRouter.get('/', use(getProducts));
 productsRouter.post('/', use(createProduct));
 productsRouter.put('/:id', use(uploadProductImage));
 productsRouter.use(['/popular', '/limited-edition'], paginateMiddleware);
 productsRouter.get('/popular', use(getPopularInTheCommunity));
+productsRouter.get('/search', use(searchProducts));
```

Comment:
The best practice is simpler than this, it could be like:
```
productsRouter.get('/popular', paginateMiddleware, use(searchProducts));
```

And the same way for each api. So you can know the flow for each path clearly